bindings/python/src/execution.rs

```diff
@@ -0,0 +1,224 @@
+use std::any::Any;
+use std::fmt::Debug;
+use std::sync::{Arc, Mutex};
+
+use arrow_util::pretty;
+use async_trait::async_trait;
+use datafusion::arrow::datatypes::{Schema, SchemaRef};
+use datafusion::arrow::pyarrow::ToPyArrow;
+use datafusion::datasource::TableProvider;
+use datafusion::execution::context::SessionState;
+use datafusion::execution::TaskContext;
+use datafusion::logical_expr::{TableProviderFilterPushDown, TableType};
+use datafusion::physical_plan::stream::RecordBatchStreamAdapter;
+use datafusion::physical_plan::streaming::{PartitionStream, StreamingTableExec};
+use datafusion::physical_plan::ExecutionPlan;
+use datafusion::prelude::Expr;
+use futures::StreamExt;
+use glaredb::{DataFusionError, Operation, RecordBatch, SendableRecordBatchStream};
+use pyo3::exceptions::PyRuntimeError;
+use pyo3::prelude::*;
+use pyo3::types::PyTuple;
+
+use crate::error::PyGlareDbError;
+use crate::runtime::wait_for_future;
+use crate::util::pyprint;
+
+#[pyclass]
+#[derive(Debug, Clone)]
+pub struct PyExecution {
```

Comment:
what is a `PyExecution`? I don't understand the naming. Usually when creating binding wrapper structs, they are transparent

```rs
pub struct MyStruct {}

pub struct PyMyStruct {
  inner: MyStruct // or Arc<MyStruct>
}
```

```suggestion
pub struct PyOperation {
```

Reply:
I'm not sure that operation is a good name either, though none of this naming is exposed to users.

Reply:
in duckdb's python api, this is referred to as [DuckDBPyRelation](https://duckdb.org/docs/api/python/reference/#duckdb.DuckDBPyRelation). maybe we could just go with that? It seems to follow the same semantics


> Run a SQL query. If it is a SELECT statement, create a relation object from the given SQL query, otherwise run the query as-is.

Reply:
Relation makes about as much sense as LogicalPlan, which is to say, not much unless you know what's happening under the covers. I opted for `ExecutionOutput` in the latest version. 

Previously the bindings had this `PyExecutionResult` and I think having Result in that name is misleading, and also it's not a result as much as a promise... So I got `PyExecution` from there, which works-ish...